backend/app/agents/shared/base_agent.py

```
@@ -0,0 +1,74 @@
+
+    def _build_graph(self):
+        """Build the LangGraph workflow - to be implemented by subclasses"""
+        raise NotImplementedError("Subclasses must implement _build_graph")
+
+    async def run(self, initial_state: AgentState) -> AgentState:
+        """Execute the agent workflow"""
+        try:
+            logger.info(f"Starting {self.agent_name} for session {initial_state.session_id}")
+            result = await self.graph.ainvoke(initial_state.model_dump())
+            return AgentState(**result)
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)